src/Core/Resettable.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace SilverStripe\Core;
+
+/**
+ * Represents a class with a local cache which normally invalidates itself between requests.
+ *
+ * Designed so that tests can automatically flush these objects between runs in lieu
+ * of a real change.
+ */
+interface Resettable
```

Comment:
My thinking is that we should introduce some kind of `Application` object that contains any in-memory caches and other objects that we might want to re-set, and have our tests create new Application objects. But it's a separate PR.